### PR TITLE
Add Workplace Search output

### PR DESCRIPTION
**Pull request by davishmcclurg**
This is mostly copied from the app search output with some changes for
the API and client differences.

I tried to switch to the [shared client][0] but that resulted in a
dependency conflict on `elasticsearch-transport`.

The integration spec is using internal Workplace Search APIs because we
don't currently expose enough to make it work with public APIs (can't
get access token or content source ID). Hopefully this will change in
the near future.

[0]: https://github.com/elastic/enterprise-search-ruby

Extracted from: https://github.com/logstash-plugins/logstash-output-elastic_app_search/pull/26

### CODE REVIEW

**Review by andsel — CHANGES_REQUESTED** [review]
This PR is almost ok, but need some touches to the documentation part:
-  comment all config options in https://github.com/logstash-plugins/logstash-integration-elastic_enterprise_search/pull/4/files#diff-25818597a9eb4becce95b3d4c2ae759cf0be5cdcd3dc2313c3913489c4ddddd0R8-R12
- align the documentation file `docs/output-elastic_workplace_search.asciidoc` to contain the description of what this plugin does and describe the configuration options. Actually it's just a copy&paste of the existing `docs/output-elastic_app_search.asciidoc`

**Comment by andsel** [comment]
@davishmcclurg the shared client introduction is drafted in PR #3 but that should be published only after Logstash `7.14` is out, so this PR is safe to be released when this PR is merged

**Comment by davishmcclurg** [comment]
@andsel I updated the docs: c7892b6fda49c3ef18c7585b6cbf6d84f606b133

I'm not sure what you mean by commenting the config options. Is that something that's missing from the app search output as well? https://github.com/logstash-plugins/logstash-integration-elastic_enterprise_search/blob/164e13f37f12f2dc226b28485f66e9c1409f40eb/lib/logstash/outputs/elastic_app_search.rb#L8-L14

**Comment by andsel** [comment]
Uhh sorry @davishmcclurg, my bad I didn't check it, usually also in code the documentation string is replicated, it's an habit, but also App Search Output hasn't,  I fix it

**Comment by andsel** [comment]
Since we have some additions to the documentation (`docs/output-elastic_workplace_search.asciidoc`), I would ask an eye to @karenzone

**Review by andsel — APPROVED** [review]
the code part LGTM, wait for review on docs

**Comment by davishmcclurg** [comment]
> usually also in code the documentation string is replicated, it's an habit, but also App Search Output hasn't, I fix it

Thanks for adding that @andsel!

File: docs/output-elastic_workplace_search.asciidoc
Comment (karenzone) [inline docs/output-elastic_workplace_search.asciidoc:20]:
```suggestion
include::{include_path}/plugin_header-integration.asciidoc[]
```

File: docs/output-elastic_workplace_search.asciidoc
Comment (karenzone) [inline docs/output-elastic_workplace_search.asciidoc:22]:
```suggestion
[id="plugins-{type}s-{plugin}-description"]
==== Description
```

File: docs/output-elastic_workplace_search.asciidoc
Comment (karenzone) [inline docs/output-elastic_workplace_search.asciidoc:98]:
```suggestion
```

**Review by karenzone — COMMENTED** [review]
Thanks for adding docs, @davishmcclurg. I left some comments inline for your consideration.

File: docs/index.asciidoc
Comment (karenzone) [inline docs/index.asciidoc:28]:
```suggestion
- Workplace Search output plugin
```

Comment (karenzone) [reply docs/index.asciidoc:28]: The generated file doesn't exist yet, and linking to it will break the build. Let's hard code it for now, and I will add back the link when I hook up the other doc pieces.

**Review by karenzone — APPROVED** [review]
LGTM! We can merge, but please hold off on publishing the gem. I have a couple of other changes and fixes I'd like to get in for this integration.

**Comment by karenzone** [comment]
We can merge, but please hold off on publishing. I'd like to include a few more docs fixes, and then carefully monitor post-publish to avoid introducing problems that would bring down the elastic.co docs build.

File: docs/index.asciidoc
Comment (davishmcclurg) [reply docs/index.asciidoc:28]: Should I make this change before merging?

Comment (karenzone) [reply docs/index.asciidoc:28]: It won't break the build until after we publish the gem, but I'm always hesitant to merge time bombs. If this is the only thing preventing you from merging, please make this change and I'll add your link back as part of a PR I have inflight now.

Comment (davishmcclurg) [reply docs/index.asciidoc]: Removed!

**Comment by karenzone** [comment]
Related PRs discussed in comments:
* #5 to fix app_search doc (ideally we can bundle with this one and publish together)
* #6 to turn on links for plugin docs (can't be merged until after #5 has been merged and the gem published to enable docgen)

**Comment by davishmcclurg** [comment]
@andsel does this look ready to merge to you? I don't think I have the power.

**Comment by andsel** [comment]
@davishmcclurg for me it's ok, for @karenzone also, we could merge after the #5 is merged  so that at that point we can also publish in the same round.

**Comment by karenzone** [comment]
@andsel @davishmcclurg  I just merged #5 so it can be included when we publish.

**Comment by karenzone** [comment]
Ugh... We broke the docs build. :-(. Somehow the commit associated with this change got stomped:
https://github.com/logstash-plugins/logstash-integration-elastic_enterprise_search/pull/4#discussion_r632203061. I will fix to get the docs builds going again.

**Pull request merged**
